wgpu-core/src/track/mod.rs

```diff
@@ -607,9 +605,11 @@ impl<A: hub::HalApi> UsageScope<A> {
         textures: &hub::Storage<resource::Texture<A>, id::TextureId>,
         bind_group: &BindGroupStates<A>,
     ) -> Result<(), UsageConflict> {
-        self.buffers.merge_bind_group(&bind_group.buffers)?;
-        self.textures
-            .merge_bind_group(textures, &bind_group.textures)?;
+        unsafe { self.buffers.merge_bind_group(&bind_group.buffers)? };
+        unsafe {
+            self.textures
+                .merge_bind_group(textures, &bind_group.textures)?
+        };
```

Comment:
Combine lke here

Reply:
Combined in 0acce788.

Reply:
@cwfitzgerald: I've tried to find more cases where your suggestion may apply; how does 67817290/overall look?